kore/src/Kore/Step/Axiom/Evaluate.hs

```diff
@@ -68,7 +72,7 @@ evaluateAxioms
     -> simplifier (Step.Results variable)
 evaluateAxioms definitionRules termLike predicate
   | any ruleIsConcrete definitionRules
-  , not (TermLike.isConcrete termLike)
+  , not (all TermLike.isNonSimplifiable termLikeF)
```

Comment:
This looks a little strange. If I understand correctly, since this is supposed to be a Function application, and since that is `Foldable`, what `all` will do would be to iterate over the arguments. Is that correct?  If so, although quite clever, it does not seem straight-forward to read, since `TermLikeF` being `Foldable` is not something we currently use on a regular basis in the code.  Therefore it might be useful to accompany this change by a comment.

Reply:
Better now?

Reply:
better by why `isConstructorLike` instead of `isNonSimplifiable`?

Reply:
Because @ttuegel just renamed it ;)

Reply:
https://github.com/kframework/kore/pull/1322